examples/SmoothedParticleHydrodynamics/src/model/functions.c

```diff
@@ -0,0 +1,233 @@
+
+/*
+ * FLAME GPU v 1.5.X for CUDA 9
+ * Copyright University of Sheffield.
+ * Original Author: Dr Paul Richmond (user contributions tracked on https://github.com/FLAMEGPU/FLAMEGPU)
+ * Contact: p.richmond@sheffield.ac.uk (http://www.paulrichmond.staff.shef.ac.uk)
+ *
+ * University of Sheffield retain all intellectual property and
+ * proprietary rights in and to this software and related documentation.
+ * Any use, reproduction, disclosure, or distribution of this software
+ * and related documentation without an express license agreement from
+ * University of Sheffield is strictly prohibited.
+ *
+ * For terms of licence agreement please attached licence or view licence
+ * on www.flamegpu.com website.
+ *
+ */
+
+/* 
+	Smoothed Particle Hydrodynamics Demo
+	Author: Matthew Leach
+	Contact: M.Leach@sheffield.ac.uk
+
+	Suggested things to try:
+	
+	Play with the viscosity to try fluids of different thickness - the simulation is
+	stable for viscosity values between ~0.1 - 30
+
+	Lower the fluid rest density to TODO see the 'FLAME' explode instead of fall
+
+	Use the python script in the iterations folder to change the initial arrangement of the fluid particles
+*/
+
+
+#ifndef _FLAMEGPU_FUNCTIONS
+#define _FLAMEGPU_FUNCTIONS
+
+#include <header.h>
+
+// Mathematic constants
+#define PI 3.141593f
+
+// Sim parameters
+#define TIMESTEP 0.001f
```

Comment:
you could make these environment variables which would support parameterisation within `0.xml`, so they are automatically loaded from 0.xml and can be configured without recompilation (if appropriate). 

The default value(s) can be specified in `XMLModelFile.xml`

In device code the symbol is the same, but if accessed in host code you'd have to use the `float* get_TIMESTEP()` methods (and `set_TIMESTEP(float* timestep_ptr)` if required).

It doesn't make sense to do this for PI however as that is a universal constant (could use glm's PI or math.h's M_PI instead if you prefer, again this is optional).